core/shared/platform/windows/win_clock.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (C) 2023 Amazon Inc.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include "platform_api_vmcore.h"
+#include <winternl.h>
+
+#define NANOSECONDS_PER_SECOND 1000000000ULL
+#define NANOSECONDS_PER_TICK 100
+
+static int
+calculate_monotonic_clock_frequency(uint64 *outfrequency)
+{
+    LARGE_INTEGER frequency;
+    if (!QueryPerformanceFrequency(&frequency)) {
+        return BHT_ERROR;
+    }
+    else {
+        *outfrequency = (uint64)frequency.QuadPart;
+        return BHT_OK;
+    }
+}
+
+// The implementation below dervies from the following source:
+// https://github.com/WasmEdge/WasmEdge/blob/b70f48c42922ce5ee7730054b6ac0b1615176285/lib/host/wasi/win.h#L210
+static uint64
+from_file_time_to_wasi_timestamp(FILETIME filetime)
+{
+    static const uint64 ntto_unix_epoch =
+        134774ULL * 86400ULL * NANOSECONDS_PER_SECOND;
+
+    ULARGE_INTEGER temp = { .LowPart = filetime.dwLowDateTime,
+                            .HighPart = filetime.dwHighDateTime };
+
+    return temp.QuadPart - ntto_unix_epoch;
+}
+
+uint64
```

Comment:
```suggestion
int
```
Should that be `int` instead? We return `BHT_ERROR`/`BHT_OK` here.